dcpy/test/lifecycle/package/resources/proposed_md/proposed-metadata.yml

```diff
@@ -0,0 +1,96 @@
+name: db-colp
+display_name: City Owned and Leased Property (COLP)
+summary: "Raw Summary"
+description: "Raw Description"
+each_row_is_a: ""
+tags: ["a", "b"]
+
+columns:
+  - name: uid # key? id? or leave it?
+    display_name: Unique ID
+    data_source: Department of City Planning
+    description: ""
+    data_type: text
+    example: cbe20732be28f6ab445289d7a67bb241
+    checks:
+      is_primary_key: True
+    custom:
+      # not broadly applicable, so chuck under "custom"
+      readme_data_type: string
+
+assembly:
+  - key: csv_package
+    type: zip
+    filename: csv_package.zip
+    contains:
+      - key: primary_csv
+        filename: colp_{{ version }}.csv
+      - key: colp_readme
+        filename: colp_readme.pdf
+
+package_files:
+  - key: primary_shapefile
+    filename: colp_single_feature_shp.zip
+    type: shapefile
+    overrides:
+      columns:
+        GEOM:
+          name: geometry
+          data_type: geom_point
+      display_name: "display_name overridden at the dataset_file level"
+      description: "description overridden at the dataset_file level"
+    custom:
+      ignore_validation: ["AGREEMENT", "DCPEDITED", "FINALCOM"]
+
+  - key: primary_csv
+    filename: colp_single_feature.csv
+    type: csv
+    overrides:
+      omit_columns: ["GEOM"]
+
+  - key: colp_readme
+    filename: colp_readme.pdf
+
+# Change destinations to be same between different types (e.g. Socrata and Bytes)
+# Adding new destination types shouldn't require a code change in our models.
+# TODO: how to maintain some static checking
+destinations:
+  - id: socrata_prod # or key? or do we like `id`
+    type: socrata
+    custom:
+      four_four: "fn4k-qyk2"
+    files:
+      - id: primary_shapefile
+        overrides:
+          omit_columns: [dcpedited, uid]
+        custom:
+          display_name: "display_name overridden at the destination level"
+          destination_use: dataset_file # better name than type? destination_use
```

Comment:
I'm having some trouble naming this concept... but what I'm going for is, suppose we have a CSV in a dataset. That csv might be a `dataset_file` in one destination and an attachment on Socrata. My gut was to use `destination_type` but is that overloading the word `type`?